### ISSUE TITLE
LifecycleNode publishes a null timestamp in `transition_event` topics

### ISSUE THREAD

**Issue by Myzhar** [ros2/rclcpp#564]
**Required Info:**

- Operating System:
  - Ubuntu 16.04
- Installation type:
  - Source
- Version or commit hash:
  - eb439ddc73c7f57ef59bb4919b02c4dd00d6ffec
- DDS implementation:
  - Fast-RTPS
- Client library (if applicable):
  - rclcpp

#### Steps to reproduce issue
1. start a lifecycle talker node: 
   `$ ros2 run lifecycle lifecycle_talker`
2. subscribe to the `transition_event` topic:
   `$ ros2 topic echo /lc_talker/transition_event`
3. require some transitions:
   `$ ros2 lifecycle set /lc_talker configure`
   `$ ros2 lifecycle set /lc_talker activate`

I noticed the issue with my own lifecycle node and I tested it with the provided example to be sure that the issue is generic.

#### Expected behavior
current timestamp for the messages
#### Actual behavior
The timestamp is null as you can see in the following screenshoot

![image](https://user-images.githubusercontent.com/3648617/46523267-84dbc100-c885-11e8-9c58-094a9a1d58ff.png)

**Comment by Myzhar** [comment]
I noticed that also the `transition` field is empty.
Only the `ßtart-state` and the `goal-state` fields are filled

**Comment by clalancette** [comment]
Yeah, this is because the low-level publisher only fills in `start-state` and `goal-state` here: https://github.com/ros2/rcl/blob/master/rcl_lifecycle/src/com_interface.c#L229 .  This could probably be improved, so I'm going to mark this as an enhancement.

**Comment by SuperJappie08** [comment]
This is still the case in Jazzy. Related (ros2/rcl#1019, ros2/rcl#1140)

**Comment by LastStarDust** [comment]
I just came here to say that this 

> I noticed that also the transition field is empty.
> Only the ßtart-state and the goal-state fields are filled

is still the case in Jazzy but @SuperJappie08 beat me to it

**Comment by SuperJappie08** [comment]
> is still the case in Jazzy but [@SuperJappie08](https://github.com/SuperJappie08) beat me to it

@LastStarDust, I wanted to let you know that this will stay broken in Jazzy and Kilted, as the required changes are ABI breaking.
Which makes them not backportable.

It should be fixed in Lyrical (upcoming ROS 2 L-Turtle) and in the next Rolling sync.

**Comment by LastStarDust** [comment]
Thank you for letting me know. Our code logic does not depend on it, so not a big deal for us.
